katdal/dataset.py

```diff
@@ -822,7 +832,12 @@ def select(self, **kwargs):
             elif k == 'ants':
                 ants = _selection_to_list(v)
                 ant_names = [(ant.name if isinstance(ant, katpoint.Antenna) else ant) for ant in ants]
-                self._corrprod_keep &= [(inpA[:-1] in ant_names and inpB[:-1] in ant_names)
+                if self._is_deselection(ant_names):
```

Comment:
I prefer this to a separate method (especially a static one):
```Python
is_deselection = set(name[0] for name in ant_names) == {'~'}
```

Reply:
I do want us to use the same is deselection throughout the select method, so I do want it to exist as a function. 
If you feel it is too small for a method, then how about a function or even a lambda within the scope of this method?

Reply:
A function is fine (see e.g. `_selection_to_list`). My issue is that `self` is not involved so there is no need to have it on the class.